Agents/BuildingMatchingAgent/BuildingMatchingAgent/src/main/java/uk/ac/cam/cares/jps/BuildingMatchingAgent.java

```diff
@@ -0,0 +1,361 @@
+
+                            Polygon polygon = getPolygon(data, datatypeIri);
+                            centroid = computeCentroid(polygon.getExteriorRing().getCoordinates(), true);
+                            Double area = polygon.getArea();
+                            xsum += centroid.x * area;
+                            ysum += centroid.y *area;
+                            zsum += centroid.z *area;
+                            asum += area;
+                        }
+//                        lat_lon = new Coordinate(xsum/asum, ysum/asum, zsum/asum);
+                        lat_lon = transformEPSG27700ToWGS84(new Coordinate(xsum/asum, ysum/asum, zsum/asum));
+                    }
+                    else {
+                        String data = geom_result.getJSONObject(0).getString(GEOMETRY_TYPE);
+                        String datatypeIri = geom_result.getJSONObject(0).getString(DATATYPE);
+                        Polygon polygon = getPolygon(data, datatypeIri);
+                        centroid = computeCentroid(polygon.getExteriorRing().getCoordinates(), true);
+//                        lat_lon = centroid;
+                        lat_lon = transformEPSG27700ToWGS84(centroid);
+                    }
+                    StringBuilder value = new StringBuilder();
+                    for (int i = 1; i >= 0; i--)
+                        value.append("#").append(lat_lon.getOrdinate(i));
+                    value.deleteCharAt(0);
+                    triple = new Triple(NodeFactory.createURI(env_bldg), NodeFactory.createURI(kbUri + "hasWGS84LatitudeLongitude"), NodeFactory.createLiteral(value.toString(), new BaseDatatype("http://www.bigdata.com/rdf/geospatial/literals/v1#lat-lon")));
+                    insertion_centroid.addInsert(triple);
+                } catch (ParseException | CRSException | CoordinateOperationException | IllegalCoordinateException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        obe_client.executeUpdate(new UpdateRequest().add(insertion.build()).toString());
+        obe_client.executeUpdate(new UpdateRequest().add(insertion_centroid.build()).toString());
+    }
+
+    private static SelectBuilder ocgmlQueryBuilder() throws ParseException {
+//        WhereBuilder where = new WhereBuilder().addPrefix(KEY_OCGML, ocgmlUri).addWhere(QM+BLDG, KEY_OCGML+":objectClassId",  "26");
+//
+//        return new SelectBuilder().setDistinct(true)
+//                .addPrefix(KEY_OCGML, ocgmlUri).addPrefix("osid", osidUri)
+//                .addVar(QM+BLDG).addVar(QM+uprn)
+//                .addGraph(NodeFactory.createURI(bldgGraph), where)
+//                .addBind("IRI(REPLACE(str("+QM+BLDG+"), \"building\", \"cityobject\"))", QM+CITYOBJ)
+//                .addWhere(QM+ATTR, "osid:intersectsFeature" , QM+CITYOBJ)
+//                .addWhere(QM+ATTR, "osid:hasValue", QM+uprn).addOrderBy(QM+BLDG);
```

Comment:
I'm guessing this commented-out bit is just for testing/debugging? If so, please remove.

Reply:
Addressed by commit [00b2e50](https://github.com/cambridge-cares/TheWorldAvatar/pull/616/commits/00b2e50c4103dbf69dfc04feb670fbe5c9fd1617)